File: test-agent/todo-cli/manager.py
from storage import load_todos, save_todos


def add_todo(title):
    todos = load_todos()
    todo = {
        "id": len(todos) + 1,
        "title": title,
        "done": False,
    }
    todos.append(todo)
    save_todos(todos)
    return todo

File: test-agent/todo-cli/manager.js
const { loadTodos, saveTodos } = require("./storage");

function addTodo(title) {
  const todos = loadTodos();
  const todo = {
    id: todos.length + 1,
    title,
    done: false,
  };
  todos.push(todo);
  saveTodos(todos);
  return todo;
}

function getTodos() {
  return loadTodos();
}

function toggleDone(id) {
  const todos = loadTodos();
  const todo = todos.find((t) => t.id === id);
  if (!todo) return null;
  todo.done = !todo.done;
  saveTodos(todos);
  return todo;
}

module.exports = { addTodo, getTodos, toggleDone };
